tests/cli/test_datasets.py

```diff
@@ -1547,6 +1547,38 @@ def test_dataset_tag(tmpdir, runner, project, subdirectory):
     assert 0 == result.exit_code, format_result_exception(result)
 
 
+def test_dataset_overwrite_tag(tmpdir, runner, project):
+    """Test that dataset tags can be overwritten."""
+    result = runner.invoke(cli, ["dataset", "create", "my-dataset"])
+    assert 0 == result.exit_code, format_result_exception(result)
+    assert "OK" in result.output
+
+    # create some data
+    new_file = tmpdir.join("file")
+    new_file.write("test")
+
+    # add data to dataset
+    result = runner.invoke(cli, ["dataset", "add", "--copy", "my-dataset", str(new_file)], catch_exceptions=False)
+    assert 0 == result.exit_code, format_result_exception(result)
```

Comment:
We can use the `project_with_datasets` fixture that already includes 2 datasets and some data files.